Pilot/DBController.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  DBController.swift
+//  Pilot
+//
+//  Created by Nick Eckert on 7/31/16.
+//  Copyright © 2016 Sanction. All rights reserved.
+//
+
+import Cocoa
+import RealmSwift
+
+class DBController: NSObject {
+
+  static let sharedDBController = DBController()
+
+  var realm: Realm!
+
+  // Initialzie DBController by attempting to create connection to DB
+  override init() {
+    do {
+      realm = try Realm()
+    } catch {
+      // Throw appropriate error later
+    }
+  }
+
+  // Create a facebookFile to store in realmDB
+  func createFacebookFile(file: CloudFile, completion: (LocalFile) -> ()) {
+    do {
+      try realm.write {
+        let fbPhoto = FacebookFile()
+        fbPhoto.name = file.name
+        fbPhoto.fileTypeRaw = file.fileType.rawValue
+        fbPhoto.height? = file.height!
+        fbPhoto.width? = file.width!
+
+        realm.add(fbPhoto)
+
+        // Construct a LocalFile to be returned upon completion
+        let returnFile = LocalFile(name: file.name, fileType: file.fileType)
+        completion(returnFile)
+      }
+    } catch {
+        // Throw appropriate error
+    }
+  }
+
+  // Update a file stored in realmDB by providing a name
+  func updateFacebookFile(file: CloudFile) {
+
+  }
+
+  // Delete a file stored in realmDB by providing a name
+  func deleteFacebookFileByName(name: String) {
+    if let fileToDelete = getFacebookFileByName(name) {
+      do {
+        try realm.write {
+          realm.delete(fileToDelete)
+        }
+      } catch {
+        print("File failed to delete!")
+        // Throw error later
+      }
+    }
+  }
+
+  // Query a file stored in realmDB by providing a name
+  func getFacebookFileByName(name: String) -> FacebookFile? {
+    // Create the appropriate predicate for filtering a fesult by name
+    let predicate = NSPredicate(format: "name = %@", name)
+
+    return realm.objects(FacebookFile.self).filter(predicate).first
+  }
+
+}
+
+class FacebookFile: Object {
```

Comment:
Should this class be in the same Swift file as `DBController`? Just wondering what the reasoning is, because I assume we'll have multiple of these e.g. `TwitterFile`, `DropboxFile`.